### DISCUSSION TITLE
Passing atrributes to a link node?

### DISCUSSION THREAD

**Discussion by 4141done** [markdoc/markdoc#401] [Q&A] [answered]
Hello, probably a pretty common ask but searching hasn't brought up anything.

I've been pleasantly surprised by the ability to pass attributes to standard MarkDown tags like this
```markdown
# Header {% foo="bar" %}
```

However it seems like it's not possible for inline elements like link syntax? The below does not work:
```markdown
[You can do anything](https://zombo.com/) {% newTab=true %}
```

I've made it work by creating a custom `link` component but I'd prefer to use the standard syntax with the abiltiy to pass an attribute or two.  Is this possible?

Abridged code for reference:

```javascript
const markdocConfig = {
  nodes: {
    link: {
      ...nodes.link,
      transform(node: Node, config: Config) {
        const children = node.transformChildren(config)
        return new Tag('Link', {
          href: node.attributes.href,
          title: node.attributes.title,
          newTab: !!node.attributes.newTab,
          content: children
        })
      }
    }
  },
}
```

```javascript
type LinkProps = {
  href: string
  title?: string
  newTab: Boolean
}

export default function Link({ href, title, newTab, content }: LinkProps) {
  const targetValue = newTab ? "_blank" : undefined
  const relValue = newTab ? "noopener noreferrer" : undefined

  return (
    <a href={href} target={targetValue} rel={relValue} title={title}>{content}</a>
  )
}
```

**Comment by rpaul-stripe** [comment, marked as answer]
Yes, currently inline attributes always apply to the nearest enclosing block-level element. We have some thoughts about how to potentially support custom attributes on links and images, but we haven't really settled on a syntax yet. There's an open ticket about that here: https://github.com/markdoc/markdoc/issues/156

Using a custom tag, as you have done above, is currently the approach that we recommend for this until we have a real solution.

EDIT: I just posted a comment on that issue thread with some syntax proposals. Feel free to provide feedback if you have any thoughts about how this should work.

**Reply by 4141done** [reply]
Thank you for taking the time to answer and give that valuable context.  I'll bop over to the link you mentioned and give my two cents.  Best!
